### PR TITLE
CSPL-892: Documenting system resources & storage requirements

### PR DESCRIPTION
**Pull request by kumarajeet**
Contains Changes for CSPL 892 and CSPL 829
   --- System resources (CPU/Mem) guidelines
   --- Storage guidelines'
   --- Supported platforms

### CODE REVIEW

**Review by kumarajeet — COMMENTED** [review]
Ed, Thanks for reviewing the changes and updating where needed!!

File: docs/StorageClass.md [hunk 1]
```diff
@@ -73,6 +88,9 @@ This will mount local, ephemeral volumes for `/opt/splunk/etc` and
 forever whenever the Pod is removed from a node (for any reason). We strongly
 discourage and do not support using this for any production environments.*
 
+## Storage Space Considerations
+
+Please see Storage guidelines at https://github.com/splunk/splunk-operator/blob/develop/docs/CustomResources.md
```
Comment (eddikay) [inline docs/StorageClass.md:93]: I don't see the intent here. What would a reader learn on that page that isn't already defined here?

**Review by eddikay — CHANGES_REQUESTED** [review]

File: docs/StorageClass.md (hunk 1 above)
Comment (kumarajeet) [reply docs/StorageClass.md]: Removed this paragraph and added SmartStore para in the Readme for a better access to the smartstore feature benefit

**Review by gaurav-splunk — APPROVED** [review]

**Review by eddikay — APPROVED** [review]

File: docs/README.md [hunk 2]
```diff
@@ -30,28 +30,79 @@ If you're interested in contributing to the SOK open source project, review the
...
 information.
 
+The Kubernetes infrastructure must have access to storage that meets or exceeds the recommendations provided in the the Splunk Enterprise [Reference Hardware documentation](https://docs.splunk.com/Documentation/Splunk/latest/Capacity/Referencehardware#What_storage_type_should_I_use_for_a_role.3F). 
+
+### Use of Splunk SmartStore
+As a splunk deployment's data volume increases, demand for storage typically outpaces demand for compute resources. [Splunk's SmartStore Feature](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/AboutSmartStore) allows you to manage your indexer storage and compute resources in a ___cost-effective___ manner by scaling those resources separately. SmartStore utilizes a fast, SSD-based cache on each Splunk indexer node to keep recent data locally available for search. When data rolls to WARM lifecycle stage, it is uploaded to an S3 API-compliant object store for persistence. Look into the [Configuring SmartStore](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/AboutSmartStore) document for configuring and useing SmartStore.
```
Comment (smohan-splunk) [inline docs/README.md:104]: Typo: "using" vs "useing"

**Review by smohan-splunk — APPROVED** [review]

**Review by akondur — APPROVED** [review]

File: docs/CustomResources.md [hunk 3]
```diff
@@ -69,14 +70,14 @@ The `spec` section is used to define the desired state for a resource. All
...
-| resources       | [ResourceRequirements](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#resourcerequirements-v1-core) | CPU and memory [compute resource requirements](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/) to use for each pod instance (defaults shown in example above) |
-| serviceTemplate | [Service](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#service-v1-core) | Template used to create Kubernetes [Services](https://kubernetes.io/docs/concepts/services-networking/service/) |
+| Key                   | Type       | Description                                                                                                |
+| --------------------- | ---------- | ---------------------------------------------------------------------------------------------------------- |
+| image                 | string     | Container image to use for pod instances (overrides `RELATED_IMAGE_SPLUNK_ENTERPRISE` or `RELATED_IMAGE_SPLUNK_SPARK` environment variables) |
```
[the remark this answers is not in the document]
Comment (kumarajeet) [reply docs/CustomResources.md]: Thanks for checking Sirish, it should be fixed now!

File: docs/README.md [hunk 4]
```diff
@@ -30,28 +30,79 @@ If you're interested in contributing to the SOK open source project, review the
...
+
+You can submit defects to https://github.com/splunk/splunk-operator/issues.
+
+### Kubernetes Platform Suggestions:
+
+In order to create Splunk deployments through Splunk operator, access to a functioning kubernetes environment is required. Some options include:
+
+
+
+| Platform        | Splunk Smartstore Support | Additional Info|
```
[the remark this answers is not in the document]
Comment (kumarajeet) [reply docs/README.md:51]: Add text for empty cells.

File: docs/README.md [hunk 5]
```diff
@@ -30,28 +30,79 @@ If you're interested in contributing to the SOK open source project, review the
...
+
 
 *Kubernetes releases 1.16.0 and 1.16.1 contain a
 [critical bug(https://github.com/kubernetes/kubernetes/pull/83789) that can
 crash your API server when using custom resource definitions. Do not
 attempt to run the Splunk Operator using these releases. This bug is fixed in
 Kuberenetes 1.16.2.*
 
+### Splunk Enterprise Compatibility
+Each Splunk Operator release has specific Splunk Enterprise compatibility requirements. Before installing or upgrading the Splunk Operator, review the [Change Log](#ChangeLog.md) to verify version compatibility with Splunk Enterprise releases.
```
[the remark this answers is not in the document]
Comment (kumarajeet) [reply docs/README.md]: Aahh. fixed it.

File: docs/README.md (hunk 2 above)
Comment (kumarajeet) [reply docs/README.md]: Fixed.

File: docs/README.md [hunk 6]
```diff
@@ -30,28 +30,79 @@ If you're interested in contributing to the SOK open source project, review the
...
+| [Microsoft Azure Kubernetes Service (AKS)](https://azure.microsoft.com/en-us/services/kubernetes-service/)| No| No additional info available |
+| [Red Hat OpenShift](https://www.openshift.com/)| No| No additional info available|
+| [HPE Ezmeral](https://www.hpe.com/us/en/solutions/container-platform.html)|No | No additional info available|
+
 
 *Kubernetes releases 1.16.0 and 1.16.1 contain a
 [critical bug(https://github.com/kubernetes/kubernetes/pull/83789) that can
 crash your API server when using custom resource definitions. Do not
 attempt to run the Splunk Operator using these releases. This bug is fixed in
 Kuberenetes 1.16.2.*
```
Comment (vebken-splunk) [inline docs/README.md:65]: nit: not technically related to this PR, but typo "Kuberenetes" -> "Kubernetes" 
perhaps we can address it here

**Review by kashok-splunk — APPROVED** [review]

File: docs/README.md [hunk 7]
```diff
@@ -30,29 +33,81 @@ If you're interested in contributing to the SOK open source project, review the
...
-*Kubernetes releases 1.16.0 and 1.16.1 contain a
-[critical bug(https://github.com/kubernetes/kubernetes/pull/83789) that can
-crash your API server when using custom resource definitions. Do not
-attempt to run the Splunk Operator using these releases. This bug is fixed in
-Kuberenetes 1.16.2.*
 
+### Docker requirements
 The Splunk Operator requires these docker images to be present or available to your Kubernetes cluster:
 
 * `splunk/splunk-operator`: The Splunk Operator image (built by this repository)
```
Comment (smohan-splunk) [inline docs/README.md:67]: How about "The Splunk Operator image built by this repository or the [official release](https://hub.docker.com/r/splunk/splunk-operator) (1.0.0 RC)"

File: docs/SmartStore.md [hunk 8]
```diff
@@ -1,11 +1,14 @@
...
- * Already existing indexes data should be migrated from local storage to the remote store as a pre-requisite before configuring those indexes in the Custom Resource of the Splunk Operator. For more details, please see [Migrate existing data on an indexer cluster to SmartStore](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/MigratetoSmartStore#Migrate_existing_data_on_an_indexer_cluster_to_SmartStore)
+The Splunk Operator includes a method for configuring a SmartStore remote storage volume with index support using a [Custom Resource](https://splunk.github.io/splunk-operator/CustomResources.html). The SmartStore integration is not implemented as a StorageClass. This feature and its settings rely on support integrated into Splunk Enterprise. See [SmartStore](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/AboutSmartStore) for information on the feature and implementation considerations.
+
+ * SmartStore configuration is supported on these Custom Resources: Standalone and ClusterMaster.
+ * SmartStore support in the Splunk Operator is limited to Amazon S3 & S3-API-compliant object stores only if you are using the CRD configuration for S3.
```
Comment (smohan-splunk) [inline docs/SmartStore.md:8]: Should we update as " .. if you are using the CRD configuration for S3 as described below" ?

**Review by smohan-splunk — APPROVED** [review]

**Review by pogdin — APPROVED** [review]

File: docs/SmartStore.md (hunk 8 above)
Comment (kumarajeet) [reply docs/SmartStore.md:8]: That's good point. Let me take a look.

**Review by eddikay — APPROVED** [review]

File: docs/README.md (hunk 6 above)
Comment (vebken-splunk) [reply docs/README.md]: The Kuberenetes still exists in the latest changes, however the rest of the changes look good to me and this is not related to your change.

**Review by vebken-splunk — APPROVED** [review]

**Pull request merged**